src/OpenTK/Input/GamePadConfigurationSource.cs

```diff
@@ -32,22 +32,23 @@ namespace OpenTK.Input
     struct GamePadConfigurationSource
     {
         int? map_button;
-        JoystickAxis? map_axis;
+        int? map_axis;
         JoystickHat? map_hat;
         HatPosition? map_hat_position;
 
-        public GamePadConfigurationSource(JoystickAxis axis)
+        public GamePadConfigurationSource(bool axis, int index)
```

Comment:
Neither option is brilliant IMHO.

The enum has most of the same readability problems and puts in extra unnecessary code, whilst using factory methods breaks the constructor pattern.

Maintainer's call I think, nothing is going to be pretty.